packages/blade/src/components/Drawer/Drawer.web.tsx

```diff
@@ -0,0 +1,206 @@
+  onDismiss,
+  zIndex = componentZIndices.drawer,
+  children,
+  accessibilityLabel,
+  showOverlay = true,
+  initialFocusRef,
+  testID,
+}: DrawerProps): React.ReactElement => {
+  const closeButtonRef = React.useRef<HTMLDivElement>(null);
+  const backButtonRef = React.useRef<HTMLDivElement>(null);
+
+  useVerifyAllowedChildren({
+    children,
+    componentName: 'Drawer',
+    allowedComponents: [drawerComponentIds.DrawerHeader, drawerComponentIds.DrawerBody],
+  });
+
+  const { theme } = useTheme();
+  const drawerId = useId('drawer');
+  const { drawerStack, addToDrawerStack, removeFromDrawerStack } = useDrawerStack();
+
+  const { isMounted, isVisible } = usePresence(isOpen, {
+    enterTransitionDuration: theme.motion.duration.gentle,
+    exitTransitionDuration: theme.motion.duration.xmoderate,
+    initialEnter: true,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+  const stackingLevel = drawerStack.indexOf(drawerId) + 1;
+
+  const { refs, context } = useFloating({
+    open: isMounted,
+  });
+
+  useIsomorphicLayoutEffect(() => {
+    if (isMounted) {
+      addToDrawerStack(drawerId);
+    } else {
+      removeFromDrawerStack(drawerId);
+    }
+  }, [isMounted]);
+
+  return (
+    <DrawerContext.Provider
+      value={{ close: onDismiss, closeButtonRef, backButtonRef, stackingLevel }}
+    >
+      <FloatingPortal>
+        {isMounted ? (
+          <FloatingFocusManager
+            context={context}
+            initialFocus={initialFocusRef ?? (stackingLevel >= 2 ? backButtonRef : closeButtonRef)}
+          >
+            <Box
+              position="fixed"
+              {...metaAttribute({
+                name: MetaConstants.Drawer,
+                testID,
+              })}
+              // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+              zIndex={zIndex + stackingLevel}
+            >
+              {showOverlay || stackingLevel === 2 ? (
+                <DrawerOverlay
+                  onClick={() => {
+                    onDismiss();
+                  }}
+                  className={isVisible ? SHOW_DRAWER : ''}
+                  lockScroll={true}
+                />
+              ) : null}
+              <AnimatedDrawerContainer
+                className={isVisible ? SHOW_DRAWER : ''}
+                width={{ base: '90%', s: '375px', m: '420px' }}
+                {...makeAccessible({
+                  role: 'dialog',
+                  modal: true,
+                  label: accessibilityLabel,
+                })}
+                position="fixed"
+                top="spacing.0"
+                left="100%"
+                backgroundColor="popup.background.subtle"
+                elevation="highRaised"
+                height="100%"
+                ref={refs.setFloating}
+                onKeyDown={(event) => {
+                  if (event?.key === 'Escape' || event?.code === 'Escape') {
+                    onDismiss();
```

Comment:
The floating ui dismiss hook didn't work?

Reply:
nope. It only works when reference is also handled internally but in our case consumers handle reference